### PR TITLE
VM produces objects.

### LINKED ISSUE #3272: Nim VM creates tuples instead of objects.
And that results in bad JS codegen.
```nim
type Foo = object
  a: int

proc newFoo(): Foo =
  result.a = 123

proc testFoo(f: Foo) =
  echo f.a # Accessed as a_12345. Undefined!

const f = newFoo() # Generated as {Field0:123}
testFoo(f)
```

### PR DESCRIPTION
**Pull request by yglukhov**
VM will now produce nkObjConstr nodes for objects instead of nkPar nodes.
Fixes #3272.

### CODE REVIEW

File: compiler/vmmarshal.nim [hunk 1]
```diff
@@ -36,8 +36,8 @@ proc getField(n: PNode; position: int): PSym =
 proc storeAny(s: var string; t: PType; a: PNode; stored: var IntSet)
 
 proc storeObj(s: var string; typ: PType; x: PNode; stored: var IntSet) =
```
Comment (Araq) [inline compiler/vmmarshal.nim]: Are you sure this is not used for tuples?

**Comment by yglukhov** [comment]
Added a breaking note in news.

File: compiler/vmmarshal.nim (hunk 1 above)
Comment (yglukhov) [inline compiler/vmmarshal.nim]: storeObj is called only on nodes with tyObject types. With this PR it is now conceptually wrong to treat objects as nkPar (tuple) nodes.

**Comment by Araq** [comment]
Cannot merge. :-(

**Comment by yglukhov** [comment]
Rebased.

**Pull request merged**